lib/rubocop/cop/infinum/factory_association.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Infinum
+      # This cop looks for 'association' in factories that specify a ':factory' option
+      #
+      # @example
+      #   #bad
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'sth' }
+      #        association :author
+      #      end
+      #    end
+      #
+      #    FactoryBoy.define do
+      #      factory :book do
+      #        title {'sth'}
+      #        author { association :author }
+      #      end
+      #    end
+      #
+      #    #good
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'sth' }
+      #        author { build(:author) }
+      #      end
+      #    end
+      #
+      #    # author
+      #    FactoryBot.define do
+      #      factory :author do
+      #        # defined author attributes
+      #      end
+      #    end
+      class FactoryAssociation < ::RuboCop::Cop::Cop
+        MSG = 'Use association_name { build(:factory_name) } instead'
```

Comment:
That's what I was trying to do in the first place, capture 2 things, and I managed to do it, sort of. But the problem I was having was then how to use it properly. The first problem I ran into is let's say we have 
`author { association :author }` (inline definition), for some reason on_send(node) considers `author` and `{ association :author }` as two different nodes so covering the whole `author { association :author }` seems problematic. The other thing that somehow works weirdly is this:
```ruby
        def_node_matcher :association_matcher, <<~PATTERN
          (send nil? :association (:sym $_)  (hash (pair (sym :factory) (sym $_)))?)
        PATTERN
```
Here, it will do both captures properly, but in a situation like this:
```ruby
        def_node_matcher :association_matcher, <<~PATTERN
          (send nil? :association (:sym $_)  #explicit_factory ?)
        PATTERN

        def_node_matcher :explicit_factory, <<~PATTERN
          (hash (pair (sym :factory) (sym $_)))
        PATTERN
```
It will only do capture on the first attribute.
So what I managed to do was actually with first method capture both attributes in case of using the association with a custom factory, but did not manage to apply it to the situation of `author { association :author }` because of the above written problems and because of it in the end just went with the most simple solution. I will do it the way you recommended, but would like it if you can give me some ideas on how to approach the problem of `author { association :author }` so that I can maybe implement the autocorrector and the messages to work fully

Reply:
From the top of my head, it seems to me you're missing some pattern matching operators if the behaviour is different when using 1 matcher and 2 matchers...on first glance, this change might do it for you:

```ruby
def_node_matcher :association_matcher, <<~PATTERN
  (send nil? :association (:sym $_)  $#explicit_factory ?)
PATTERN
```
or even
```ruby
def_node_matcher :association_matcher, <<~PATTERN
  (send nil? :association (:sym $_)  <$#explicit_factory ...> ?)
PATTERN
```

Unfortunately, I also worked for a couple of days on this initially, so I can't say for certain what the problem is.

Give it a couple more tries and dive in other Rubocop cops. I know there are plenty and it's a bit tough, but there must be a case similar enough to our that will help you 🙂

Reply:
So, `attribute1` becomes `bar` when it should be `foo`. This means that there's a problem with the capturing using `$`. I'd check out if there are any cops that have to capture 2 nodes. Maybe we're passing them wrong in the block... 🤔

Reply:
If you get an epiphany, don't hesitate to create a new PR with autocorrecting 🙂 
I remember that I also added fixes and enhancements to my custom cop after resting a bit from the implementation 🧠

Reply:
I got an epiphany and actually made the autocorrecto 🎉 . It's a bit hacky, but hey, it does the job. Will now make the tests for it just to test out all of it works 😄

Reply:
One thing I am a bit hating on is the fact that this is written in ruby 2.5.0 and ruby 2.5. does not have .second for arrays 😡